From 26ad9c9c1d0b97c923fce2a4e1f0c9375c3f9736 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Fri, 6 Jan 2023 21:00:01 +0800
Subject: [PATCH 6/6] soong: Specify heap size for metalava for S+

filiprrs: This is needed on systems with less than 16GB physical ram.
Compiling using a single job is recommended.

mnri: increase heap size to 8gb, reference: e5887b3107cffab2e208ff94a93284cae3a5fd1f
      update commit title and description

test: mka bacon -j8, no OOM encountered on 15GB ram machine

Signed-off-by: Michael W <baddaemon87@gmail.com>

Change-Id: I5093dcdbe384fd33f0dc0fd9b89c91f8f9fa19fd
---
 java/droiddoc.go  | 1 +
 ui/build/build.go | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/java/droiddoc.go b/java/droiddoc.go
index b0d537617..c00d27df4 100644
--- a/java/droiddoc.go
+++ b/java/droiddoc.go
@@ -706,6 +706,7 @@ func javadocCmd(ctx android.ModuleContext, rule *android.RuleBuilder, srcs andro
 	cmd := rule.Command().
 		BuiltTool("soong_javac_wrapper").Tool(config.JavadocCmd(ctx)).
 		Flag(config.JavacVmFlags).
+		Flag("-J-Xmx8G").
 		FlagWithRspFileInputList("@", android.PathForModuleOut(ctx, "javadoc.rsp"), srcs).
 		FlagWithInput("@", srcJarList)
 
diff --git a/ui/build/build.go b/ui/build/build.go
index 9a9eccd7d..affd10ba3 100644
--- a/ui/build/build.go
+++ b/ui/build/build.go
@@ -194,6 +194,10 @@ func checkRAM(ctx Context, config Config) {
 			ctx.Println("************************************************************")
 			ctx.Printf("You are building on a machine with %.3vGB of RAM\n", ram)
 			ctx.Println("")
+			ctx.Println("This is a modified instruction:")
+			ctx.Println("There is a workaround applied for lower system machines (<16GB),")
+			ctx.Println("we suggest increasing swap size to 1.5x to resolve most OOM errors.")
+			ctx.Println("Original warning from AOSP:")
 			ctx.Println("The minimum required amount of free memory is around 16GB,")
 			ctx.Println("and even with that, some configurations may not work.")
 			ctx.Println("")
-- 
2.44.0

